openapi.yaml

```
@@ -45,6 +45,126 @@ tags:
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+    post:
+      tags:
+        - application
+      description: Is a one-stop deployment method for repository charms. Only a charm
+        name is required to deploy. If argument validation fails, a list of all errors
+        found in validation will be returned. If a local resource is provided, details
+        required for uploading the validated resource will be returned.
+      operationId: addApplication
+      parameters:
+        - $ref: "#/components/parameters/ApplicationName"
+        - $ref: "#/components/parameters/Force"
+        - $ref: "#/components/parameters/DryRun"
+        - $ref: "#/components/parameters/Trust"
+        - $ref: "#/components/parameters/NumUnits"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DeployFromRepositoryArgs'
+        required: true
+      responses:
+        "200":
+          description: Detailed information about the requested application.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DeployFromRepositoryResult"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+    delete:
+      tags:
+        - application
+      description: Removes a given application.
+      operationId: deleteApplication
+      parameters:
+        - $ref: "#/components/parameters/ApplicationName"
+        - $ref: "#/components/parameters/DestroyStorage"
+        - $ref: "#/components/parameters/Force"
+        - $ref: "#/components/parameters/MaxWait"
+        - $ref: "#/components/parameters/DryRun"
+      responses:
+        "200":
+          $ref: "#/components/responses/ApplicationDestroyed"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+  /application/{name}/refresh:
+    patch:
+      tags:
+        - application
+      description: Sets the charm for a given for the application.
+      operationId: refreshApplication
+      parameters:
+        - $ref: "#/components/parameters/ApplicationName"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ApplicationSetCharm'
+        required: true
+      responses:
+        "200":
+          $ref: "#/components/responses/EmptySyncResponse"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+  /application/relate:
```

```suggestion
  /integrate
```

 I think that because `integrate` is not necessarily tied to a model, we should bring this up to the top level